lib/server.go

```diff
@@ -1013,14 +1109,20 @@ func (srv *Server) shouldVerifySignatures(header *MsgDeSoHeader, isHeaderChain b
 	if checkpointBlockInfo == nil {
 		return true, false
 	}
-	var hasSeenCheckpointBlockHash bool
-	var checkpointBlockNode *BlockNode
+	// TODO: @diamondhands - why can't we move this up in this function? It seems like we can avoid
```

Comment:
Please check out this optimization

Reply:
This seems fine to me. Probably just had it later because performance wasn't an issue before.